g1ant-language/language-basics.md

```diff
@@ -0,0 +1,321 @@
+
+## Language Syntax
+
+G1ANT.Language syntax is pure and simple:
+
+```G1ANT
+command argument value 
+```
+
+As you can see, you always start with a command followed by its argument and then a value of that argument. Compare this with natural language: if you say *“pass this to me”*, “pass” would be a command, “this” — an argument, “to me” — value.
+
+Now, let’s learn who is who — or rather what is what.
+
+### Command
+
+Is an instruction given by a user telling the robot to do something — for example, run a program or perform keyboard or mouse action. You could compare commands to verbs in natural language.
+
+A command can be simple, one-word, like `dialog` or extended, like `selenium.open`. Two-word commands separated with a dot in the middle belong to a family of commands. It means that there are other commands starting with `selenium.` and their suffix (the part following the dot) expresses certain activity: `open`, `newtab` ,`switch`.
+
+Each family of commands has its special ‘activities’. Once you start typing the command inside the script editor window, a list of available activities will pop up, as you can see below:
+
+![](.gitbook/assets/selenium.png)
+
+### Argument
+
+A command needs to know, what exactly it is supposed to do. So, you must provide some information — and this data is called an *argument*. For example, when you type `program`, the robot will suggest you next input in a drop-down list and the first item will be `name`. It is an argument for the `program` command followed with a name of the program to be launched. So, if you wanted the robot to open Notepad, you would write `program name notepad` — remember the “command argument value” rule?
+
+### Value
+
+The content of a field or a variable is called a *value*. It can refer to alphabetic as well as numeric data. For example, in the expression
+
+```G1ANT
+selenium.open type ‴firefox‴ url ‴google.com‴ result ♥google
+```
+
+the elements `‴firefox‴` and `‴google.com‴` are values. Note that text values are embraced with triple prime characters available by pressing **Ctrl+'** (apostrophe) keyboard shortcut or by clicking the text symbol icon on the toolbar.
+
+Values always belong to one type of structures, [variables](), and they can be any text, numbers and a mixture of both (URL address, for instance). Values are strictly related to their arguments.
+
+### Examples
+
+In the following example the golden “command-argument-value” rule is a bit twisted:
+
+```G1ANT
+text.read c:\Users\♥USERNAME\Desktop\g1ant.txt result ♥g1
+```
+
+The `text.read` command is immediately followed by a value (`c:\Users\♥USERNAME\Desktop\g1ant.txt`), not an argument. It’s because some argument names can be omitted, since they are obvious and mandatory. In this case, the missing argument name is `filename`, which is followed by a value with the name of a file to open. Then, there is the `result` argument and its `♥g1` value.
```

Comment:
arguments or argument names are the thing to discuss further

Reply:
```suggestion
The `text.read` command is immediately followed by a value (`c:\Users\♥USERNAME\Desktop\g1ant.txt`), not an argument. It’s because first arguments can be omitted, since they are obvious. In this case, the missing argument is `filename`, which is followed by a value with the name of a file to open. Then, there is the `result` argument and its `♥g1` value.
```